packages/editor/src/utils/media-upload/index.js

```diff
@@ -1,7 +1,12 @@
+/**
+ * External dependencies
+ */
+import { v4 as uuid } from 'uuid';
+
 /**
  * WordPress dependencies
  */
-import { select } from '@wordpress/data';
+import { select, dispatch } from '@wordpress/data';
```

Comment:
It's a bit unfortunate that we're using the global select/dispatch here. I think ideally, we should move this logic into a react hook and use the context registry instead or useSelect/useDispatch but that seems pre-existent, so, not blocking for this PR.

Reply:
That sounds good, can we work on that part in a follow up PR?

Reply:
yes, I think so.